### PR TITLE
feat(factory): retrieve all pairs

### LINKED ISSUE #43: factory: should we remove  all_pairs entrypoint?

### PR DESCRIPTION
**Pull request by bogdan-manole**
Closes #43

### CODE REVIEW

File: contracts/AedexV2Factory.aes [hunk 1]
```diff
@@ -42,7 +42,9 @@ contract AedexV2Factory =
 
     entrypoint all_pairs_length(): int = List.length(state.all_pairs)
 
-    entrypoint all_pairs(ix: int): IAedexV2Pair = List.get(ix, state.all_pairs)
+    entrypoint all_pairs(): list(IAedexV2Pair) = state.all_pairs
+
+    entrypoint get_nth_pair(ix: int): IAedexV2Pair = List.get(ix, state.all_pairs)
```
Comment (kenodressel) [inline contracts/AedexV2Factory.aes:47]: `get_nth_pair` seems weird, why not `get_all_pairs()` and `get_pair(idx)`? I think people should be able to read function signatures. Other opinions?

Comment (hanssv) [reply contracts/AedexV2Factory.aes:47]: `get_pair` and `get_all_pairs` sounds good to me.

Comment (bogdan-manole) [reply contracts/AedexV2Factory.aes:47]: `get_pair` already exists: gets the pair address based on the given token addresses 
https://github.com/aeternity/dex-contracts-v2/blob/5c0b0296f38d4b974cace6b2cf10e8135e505f94/contracts/AedexV2Factory.aes#L55-L58

File: contracts/AedexV2Factory.aes [hunk 2]
```diff
@@ -42,7 +42,9 @@ contract AedexV2Factory =
 
     entrypoint all_pairs_length(): int = List.length(state.all_pairs)
 
-    entrypoint all_pairs(ix: int): IAedexV2Pair = List.get(ix, state.all_pairs)
+    entrypoint get_all_pairs(): list(IAedexV2Pair) = state.all_pairs
+
+    entrypoint get_nth_pair(ix: int): IAedexV2Pair = List.get(ix, state.all_pairs)
```
Comment (ghallak) [inline contracts/AedexV2Factory.aes:47]: Why did you prefer to use `List.get` over `List.nth` in `get_nth_pair`? I noticed that the previously existing `get_pair` returns `option(IAedexV2Pair)` which is more difficult to be used in a wrong way (when given an out of bounds index).

Comment (bogdan-manole) [reply contracts/AedexV2Factory.aes:47]: - the `get_pair` entrypoint is used to get the instance of a pair based on some arbitrary 2 tokens, but also at the same time to see if that pair exists. So , `option(IAedexV2Pair)` serves both purposes
- `get_nth_pair` is used in the context of `all_pairs_length`, from which the caller should know in advance the length of the whole stored pairs. So , the failure in the case of an out of boundary index is an assumed one.

File: contracts/AedexV2Factory.aes (hunk 1 above)
Comment (kenodressel) [reply contracts/AedexV2Factory.aes:47]: get_nth_pair is fine by me in this case

**Pull request merged**